.github/workflows/trigger-ucc-bench-pr.yml

```diff
@@ -0,0 +1,51 @@
+# Trigger PR workflow in ucc-bench to test the performance impact of these changes
+# to the current baseline benchmark results, posting a comment back to the ucc PR
+# with a summary.
+# This will only be run if the PR has the label "preview-benchmark-results"
+# and will be skipped if the label is removed.
+
+# To trigger an event in the other repo, we had to install a GitHub app in the
+# ucc-bench repo, and then add the app ID/private key as secrets to this repo
+# so it could trigger the workflow. See
+# https://docs.github.com/en/issues/planning-and-tracking-with-projects/automating-your-project/automating-projects-using-actions
+# for details.
+
+name: Trigger ucc PR workflow in ucc-bench
+on:
+    pull_request:
+      branches:
+        - main
+      types:
+        - opened
+        - synchronize
+        - labeled
+        - unlabeled
+jobs:
+  trigger-benchmarks-pr:
+    runs-on: ubuntu-latest
+    # Don't run on forks or original benchmark commits
+    if: contains(github.event.pull_request.labels.*.name, 'preview-benchmark-results')
+
+    steps:
+      - name: Generate token
+        id: generate-token
+        uses: actions/create-github-app-token@v1
+        with:
+          app-id: ${{ secrets.UCC_BENCH_APP_ID }}
+          private-key: ${{ secrets.UCC_BENCH_APP_PRIVATE_KEY }}
+          owner: unitaryfoundation
+          repositories: |
+            ucc-bench
+            ucc
+
+      - name: Trigger ucc-main-pr event in ucc-bench
+        env:
+          GH_TOKEN: ${{ steps.generate-token.outputs.token }}
+        run: |
+          gh api \
+            --method POST \
+            -H "Accept: application/vnd.github+json" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            /repos/unitaryfoundation/ucc-bench/dispatches \
+            -f "event_type=ucc-main-pr" -F "client_payload[commit_hash]=${{ github.sha }}"\
+            -F "client_payload[pr_number]=${{ github.event.pull_request.number }}"
```

Comment:
This `repository_dispatch` feature is pretty cool! Glad to know about it.

Is it possible that some actor outside of this github action could mock the POST request to trigger the run? I guess the `GH_TOKEN` maybe prevents that?

Reply:
Yes that is correct. They would need to generate a token that has the proper access permissions on the ucc-bench repo. 

On the benefits side, users with that access (say with write access on the ucc-bench repo) could test that workflow via this call themselves.